File: Library/src/com/androidanimator/animation/Combinable.java
package com.androidanimator.animation;

import android.animation.TimeInterpolator;

/**
 * This interface is implemented only by animation classes that can be combined
 * to animate together.
 * 
 */
public interface Combinable {
	
	public void animate();
	public Animation setInterpolator(TimeInterpolator interpolator);
	public long getDuration();
	public Animation setDuration(long duration);
	public Animation setListener(AnimationListener listener);

}

File: Library/src/com/easyandroidanimations/library/kotlinlib/Combinable.kt
package com.easyandroidanimations.library

import android.animation.AnimatorSet

/**
 * This interface is implemented only by animation classes that can be combined
 * to animate together.
 *
 */
interface Combinable {
    fun animate()
    val animatorSet: AnimatorSet?
    fun setInterpolator(interpolator: TimeInterpolator?): Animation?
    val duration: Long
    fun setDuration(duration: Long): Animation?
    fun setListener(listener: AnimationListener?): Animation?
}